cmd/kube_create.go

```diff
@@ -0,0 +1,457 @@
+)
+
+// kubeCreateDebug represents a debug mode flag
+var kubeCreateDebug bool
+
+// kubeBootstrapStep represents a k8s instance bootstrap step
+type kubeBootstrapStep struct {
+	name    string
+	command string
+}
+
+type kubeCluster struct {
+	Name    string
+	Version string
+	Address string
+}
+
+// kubeBootstrapSteps represents a k8s instance bootstrap steps
+var kubeBootstrapSteps = []kubeBootstrapStep{
+	{name: "Instance system upgrade", command: `\
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get update && sudo -E DEBIAN_FRONTEND=noninteractive apt-get upgrade -y
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get install -y \
+	apt-transport-https \
+	ca-certificates \
+	curl \
+	golang-cfssl \
+	software-properties-common
+nohup sh -c 'sleep 5s ; sudo reboot' &
+exit`},
+	{name: "Docker Engine installation", command: `\
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+sudo add-apt-repository \
+	"deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+	$(lsb_release -cs) \
+	stable"
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get update
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get install -y docker-ce=18.06.0~ce~3-0~ubuntu
+
+cat <<EOF > csr.json
+{
+	"hosts": ["{{ .Address }}"],
+	"key": {"algo": "rsa", "size": 2048},
+	"names": [{"C": "CH", "L": "Lausanne", "O": "Exoscale", "OU": "exokube", "ST": ""}]
+}
+EOF
+
+cfssl genkey -initca csr.json | cfssljson -bare ca
+
+cfssl gencert \
+	-ca ca.pem \
+	-ca-key ca-key.pem \
+	-hostname {{ .Address }} csr.json | cfssljson -bare
+
+cat <<EOF | sudo tee /etc/docker/daemon.json
+{
+	"hosts": ["unix:///var/run/docker.sock", "tcp://0.0.0.0:2376"],
+	"tlsverify": true,
+	"tlscacert": "/etc/docker/ca.pem",
+	"tlscert": "/etc/docker/cert.pem",
+	"tlskey": "/etc/docker/key.pem",
+	"exec-opts": ["native.cgroupdriver=systemd"],
+	"storage-driver": "overlay2",
+	"log-driver": "json-file",
+	"log-opts": {
+		"max-size": "100m"
+	}
+}
+EOF
+
+sudo mv ca.pem /etc/docker/ca.pem
+sudo mv cert.pem /etc/docker/cert.pem
+sudo mv cert-key.pem /etc/docker/key.pem
+rm -f *.{csr,json,pem}
+
+sudo sed -i -re 's#^ExecStart=.*#ExecStart=/usr/bin/dockerd#' /lib/systemd/system/docker.service
+sudo systemctl daemon-reload && sudo systemctl restart docker`},
+	{name: "Kubernetes cluster node installation", command: `\
+curl -fsSL https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+cat <<EOF | sudo tee /etc/apt/sources.list.d/kubernetes.list
+deb http://apt.kubernetes.io/ kubernetes-xenial main
+EOF
+
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get update && \
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get install -y kubelet kubeadm kubectl && \
+sudo apt-mark hold kubelet kubeadm kubectl`},
+	{name: "Kubernetes cluster node initialization", command: `\
+sudo kubeadm init --pod-network-cidr=192.168.0.0/16 --kubernetes-version {{ .Version }} &&
+sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf taint nodes --all node-role.kubernetes.io/master- &&
```

Comment:
`&&` ?

Reply:
IMO it's not necessary to setup kube configuration for the `ubuntu` user, we're not supposed to use it.